### DISCUSSION TITLE
Why does this DXF file fail to read？

### DISCUSSION THREAD

**Discussion by cbvgn** [mozman/ezdxf#721] [Q&A]
![3d561d78765b34cd3a30121a1f2cc26](https://user-images.githubusercontent.com/110533047/182580921-759b8daf-2991-4c09-b2be-c85d770a708f.png)

**Comment by mozman** [comment]
Because there are a lot of crappy DXF files out there in the wild :wink:.

You can use the [recover](https://ezdxf.mozman.at/docs/drawing/recover.html) module to load such files. If this doesn't help and AutoCAD or any other reliable CAD application can't open the file, please post your DXF file here so I can analyze it.

**Reply by cbvgn** [reply]
The recovery module seems to have no effect. There are some problems in these files. Can I try to skip the problematic part when reading the files?
[A2.zip](https://github.com/mozman/ezdxf/files/9256139/A2.zip)
[222.zip](https://github.com/mozman/ezdxf/files/9256140/222.zip)

**Comment by mozman** [comment]
Well, these files are total crap and cannot even be recovered by a CAD application. The only action you can do is open an issue at the [libredwg](https://github.com/LibreDWG/libredwg) project that created these DXF files so they know they produce invalid DXF files.

**Reply by mozman** [reply]
additional info:

- File `222.dxf` has a root dictionary, but not as the first entity in the `OBJECTS` section as required
- File `A2.dxf` has an invalid empty line at line number 18395 and 18498, after fixing that same issue as in file `222.dxf`

**Reply by cbvgn** [reply]
Thank you for your reply, I will ask at the libredwg project

**Comment by mozman** [comment]
The recover module can now read file `222.dxf`.

The file `A2.dxf` has another DXF structure error: 

- HATCH: unknown edge type: 0

Valid edge types are 1-4, see DXF reference for Hatch, [Boundary Path Data](https://help.autodesk.com/view/OARX/2018/ENU/?guid=GUID-DC5215D6-E73F-4DFF-8BE9-01CA9610FAEE) tag 72

EDIT: there are "some" other errors - ezdxf audit report:

```text
   1. Issue [14] fixed.
   Recovered misplaced root dictionary

   2. Issue [213] fixed in DICTIONARY(#7).
   Removed entry "TCH_DBCONFIG" with invalid handle in DICTIONARY(#7)

   3. Issue [213] fixed in DICTIONARY(#7).
   Removed entry "TCH_DBPSCONFIG" with invalid handle in DICTIONARY(#7)

   4. Issue [213] fixed in DICTIONARY(#34).
   Removed entry "ByBlock" with invalid handle in DICTIONARY(#34)

   5. Issue [213] fixed in DICTIONARY(#34).
   Removed entry "ByLayer" with invalid handle in DICTIONARY(#34)

   6. Issue [213] fixed in DICTIONARY(#34).
   Removed entry "Global" with invalid handle in DICTIONARY(#34)

   7. Issue [202] fixed.
   Deleted DICTIONARY(#27C51) entity with invalid owner handle #27C50.

   8. Issue [202] fixed.
   Deleted DICTIONARY(#27C59) entity with invalid owner handle #27C58.

   9. Issue [213] fixed in DICTIONARY(#27C71).
   Removed entry "Standard" with invalid handle in DICTIONARY(#27C71)

  10. Issue [202] fixed.
   Deleted DICTIONARY(#27D5C) entity with invalid owner handle #27C72.

  11. Issue [202] fixed.
   Deleted DICTIONARY(#2ABFC) entity with invalid owner handle #4A.

Found 0 errors, applied 11 fixes
```
